### PR TITLE
Enable mouse cursors for TV devices

### PR DESCRIPTION
**Pull request by swift-kim**
Fixes https://github.com/flutter-tizen/flutter-tizen/issues/364.

- Add the `EnableCursor` method to `TizenWindowEcoreWl2` which is implemented for TV devices only. The method uses dlopen to invoke the `Cursor_Set_Config` function in `libvd-win-util.so`.
- Style the code in `SetTizenPolicyNotificationLevel` in a consistent way.

dlopen and dlsym should not affect the performance (launching time) noticeably.

### CODE REVIEW

**Comment by bbrto21** [comment]
Unfortunately, the mouse cursor doesn't show up on my TV. 
Is there anything else we need to enable cursor?

cc @yman-son

**Comment by yman-son** [comment]
> All the changes in this PR look good to me., but unfortunately, the mouse cursor doesn't show up on my TV.
> Is there anything else we need to enable cursor?

//TIZEN/[MAIN]/[ONEPROD_Prj]/[INT]/COMMON/Profile/platform/framework/web/chromium-efl/wrt/src/browser/tv/wrt_native_window_tv.cc
WRTNativeWindowTV::CreateMouseCursor(Evas_Object* window)
CreateMouseCursor API contains all the code for the mouse cursor.

I found the some different to get "surface" between your code and my code.
(Is this the cause?)


Would you like to test by forcibly enabling the mouse cursor?
vconftool set -t bool memory/window_system/input/force_enable_cursor 1 -f

File: shell/platform/tizen/tizen_window_ecore_wl2.cc [hunk 1]
```diff
@@ -114,6 +118,62 @@ void TizenWindowEcoreWl2::SetWindowOptions() {
...
+  }
+  eina_iterator_free(iter);
+
+  wl_surface* surface = ecore_wl2_window_surface_get(ecore_wl2_window_);
+  // The config_type 1 refers to TIZEN_CURSOR_CONFIG_CURSOR_AVAILABLE
+  // defined in the TV extension protocol tizen-extension-tv.xml.
+  if (!Cursor_Set_Config(surface, 1, nullptr)) {
+    FT_LOG(Error) << "Failed to set a cursor configuration.";
+  }
+#endif
```
Comment (yman-son) [inline shell/platform/tizen/tizen_window_ecore_wl2.cc:176]: dlclose doesn't need a call?

**Comment by bbrto21** [comment]
> vconftool set -t bool memory/window_system/input/force_enable_cursor 1 -f

I've been using this for a long time :)

**Comment by swift-kim** [comment]
> I found the some different to get "surface" between your code and my code.

@yman-son Our embedder code don't rely on Ecore Evas but directly creates a Ecore Wl2 window from scratch. Thus it already has a reference to the window handle and a wayland surface can be retrieved from it.

File: shell/platform/tizen/tizen_window_ecore_wl2.cc (hunk 1 above)
Comment (swift-kim) [reply shell/platform/tizen/tizen_window_ecore_wl2.cc:176]: You're right. I'll fix.

**Comment by yman-son** [comment]
> > vconftool set -t bool memory/window_system/input/force_enable_cursor 1 -f
> 
> I've been using this for a long time :)

I can find log in dlog when cursor setting is successful.
Please check it.

  145.059 I/VD_WIN_UTIL(P 7127, T 7127): cursor_module.c: Cursor_Set_Config(1046) > config_type as 1
  145.060 I/CURSORMGR(P  263, T  263): e_mod_cursormgr.c: _e_cursormgr_cb_cursor_set_config(344) > Cursor_Set_Config request from client with process:7127 config[1] !

File: shell/platform/tizen/tizen_window_ecore_wl2.cc (hunk 1 above)
Comment (yman-son) [reply shell/platform/tizen/tizen_window_ecore_wl2.cc:176]: There is also a call of CursorModule_Finalize() in my code~

**Comment by swift-kim** [comment]
@yman-son We can also see the log message you mentioned when the app launches.

```
I/VD_WIN_UTIL(  691): cursor_module.c: Cursor_Set_Config(1046) > config_type as 1
I/CURSORMGR(  262): e_mod_cursormgr.c: _e_cursormgr_cb_cursor_set_config(317) > Cursor_Set_Config request from client with process:691 config[1] !
```

But the cursor only appears on the screen when **force_enable_cursor** is set to true. We want to enable the cursor even if the value is set to false (the default value). Are you testing with **force_enable_cursor** turned off?

File: shell/platform/tizen/tizen_window_ecore_wl2.cc (hunk 1 above)
Comment (swift-kim) [reply shell/platform/tizen/tizen_window_ecore_wl2.cc:186]: Fixed in 58adc7d4944bc9a24b2cbc7e99f10bcc8f36fe1c.

**Comment by yman-son** [comment]
> @yman-son We can also see the log message you mentioned when the app launches.
> 
> ```
> I/VD_WIN_UTIL(  691): cursor_module.c: Cursor_Set_Config(1046) > config_type as 1
> I/CURSORMGR(  262): e_mod_cursormgr.c: _e_cursormgr_cb_cursor_set_config(317) > Cursor_Set_Config request from client with process:691 config[1] !
> ```
> 
> But the cursor only appears on the screen when **force_enable_cursor** is set to true. We want to enable the cursor even if the value is set to false (the default value). Are you testing with **force_enable_cursor** turned off?

I test vconf in false state for WRT. (ex : tving)
I will check if there is a code that disables the mouse cursor for .NET application in AUL or context-manager.

**Comment by yman-son** [comment]
> > @yman-son We can also see the log message you mentioned when the app launches.
> > ```
> > I/VD_WIN_UTIL(  691): cursor_module.c: Cursor_Set_Config(1046) > config_type as 1
> > I/CURSORMGR(  262): e_mod_cursormgr.c: _e_cursormgr_cb_cursor_set_config(317) > Cursor_Set_Config request from client with process:691 config[1] !
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But the cursor only appears on the screen when **force_enable_cursor** is set to true. We want to enable the cursor even if the value is set to false (the default value). Are you testing with **force_enable_cursor** turned off?
> 
> I test vconf in false state for WRT. (ex : tving) I will check if there is a code that disables the mouse cursor for .NET application in AUL or context-manager.


I applied the changed code with Kim's help and conducted the test on the TV.
the mouse cursor worked normally when vconf was disabled.
I don't know what environmental differences cause this kind of problem.
But what is certain is that the code of this PR is normal.

snapshot info : 
http://168.219.244.109/products/tv/official/2023/ONEMAIN/PontusM/TIZEN-ONEMAIN-PontusM-RELEASE_20220523.3/images/T-PTMAKUC-PERF/

**Comment by bbrto21** [comment]
> I applied the changed code with Kim's help and conducted the test on the TV.
> the mouse cursor worked normally when vconf was disabled.
> I don't know what environmental differences cause this kind of problem.
> But what is certain is that the code of this PR is normal.

Thank you for checking.

**Review by bbrto21 — APPROVED** [review]

**Pull request merged**